src/modules/orders.ts

```diff
@@ -5,35 +5,48 @@ import { Order } from "../../generated/schema"
 
 export namespace orders {
 
-    export function invalidateOrder(orderId: string): void {
+    export function invalidateOrder(orderId: string, timestamp: BigInt): Order {
 
         let order = Order.load(orderId)
 
         if (!order) {
             order = new Order(orderId)
-            log.warning('Order {} was not found. It was created for being invalidated', [orderId])
+            log.info('Order {} was not found. It was created for being invalidated', [orderId])
         }
 
         order.isValid = false
-        order.save()
+        order.invalidateTimestamp = timestamp
+
+        return order as Order
+    }
+
+    export function preSig(orderId: string, owner: string, timestamp: BigInt): Order {
+
+        let order = getOrCreateOrder(orderId, owner)
+
+        order.presignTimestamp = timestamp
```

Comment:
So we want to have in the timestamp always the latest `presignTimestamp`. i.e. if we do two presigns on the same orders, what date we want?

Reply:
for now we are keeping the last one. Another workaround would be adding a new "Signed" entity containing signed boolean with each timestamp. That will provide more information. Please let me know if you want me to implement that.

Reply:
> Another workaround would be adding a new "Signed" entity containing signed boolean with each timestamp. That will provide more information. Please let me know if you want me to implement that.

I think it's not needed